ansible_base/activitystream/apps.py

```diff
@@ -0,0 +1,38 @@
+from functools import partial
+
+from django.apps import AppConfig
+from django.db.models.signals import m2m_changed, post_save, pre_delete, pre_save
+
+from ansible_base.activitystream.signals import activitystream_create, activitystream_delete, activitystream_m2m_changed, activitystream_update
+
+
+def connect_activitystream_signals(cls):
+    post_save.connect(activitystream_create, sender=cls, dispatch_uid=f'dab_activitystream_{cls.__name__}_create')
+    pre_save.connect(activitystream_update, sender=cls, dispatch_uid=f'dab_activitystream_{cls.__name__}_update')
+    pre_delete.connect(activitystream_delete, sender=cls, dispatch_uid=f'dab_activitystream_{cls.__name__}_delete')
+
+    # Connect to m2m_changed signal for all m2m fields
+    for field in cls._meta.many_to_many:
+        if field.name in cls.activity_stream_excluded_field_names:
+            continue
+
+        fn = partial(activitystream_m2m_changed, field_name=field.name)
```

Comment:
If you're doing this as a partial, this advice may be applicable

> Note also that Django stores signal handlers as weak references by default, so if your handler is a local function, it may be garbage collected. To prevent this, pass weak=False when you call the signal’s [connect()](https://docs.djangoproject.com/en/5.0/topics/signals/#django.dispatch.Signal.connect).

Another alternative to doing a partial is to make your method a method in a class, and initialize the class in this code with `field.name` in the arguments. You would also need to save a reference (like from the AppConfig).

Looking at the call arguments of m2m_changed, I agree that manually saving the field name (in general) is a good idea, particularly if a model might use `through_fields` or some other voodoo.